dali/python/nvidia/dali/_utils/eager_utils.py

```diff
@@ -133,3 +133,9 @@ def classify_array_kwarg(arr):
                 return False, 'gpu', data
 
         return False, 'cpu', data
+
+
+def _slice_tensorlist(data, size, layout):
+    """ Constructs TensorList consisting of ``size`` first elements of ``data``. """
+
+    return type(data)(list(data)[:size], layout=layout)
```

Comment:
Looks good. Minor remark (but I'd say in the spirit of the function), `layout` should not be a parameter of the function but take it directly from `data`.

Reply:
done